frontend/lib/src/hostComm/types.ts

```diff
@@ -161,13 +161,8 @@ export type VersionedMessage<Message> = {
   stCommVersion: number
 } & Message
 
-export type IAllowedMessageOriginsResponse = {
-  allowedOrigins: string[]
-  useExternalAuthToken: boolean
-}
-
 export type IHostConfigResponse = HostConfig & {
-  allowedOrigins?: string[]
+  allowedOrigins: string[]
```

Comment:
nit: not 100% sure if we should make these properties required or optional

Reply:
Hm what's the reason to make it optional? I was thinking because the other allowed-message-origins response had this required and we are combining, we should make it required. 🤔

Reply:
The reasoning here is the following: we will likely add additional configurations to this endpoint in the future. But if we make all these options as required (meaning that they have to exist in the returned JSON), the hosts always would be forced to add the new options before a new Streamlit options get rolled out (otherwise the apps would break). This creates a lot of potential trouble for us and the host platforms. The alternative is that we make all options as optional, meaning that we use a default value if they are not set in the host config response. This is basically how our `config.toml` works, we don't require users to add values for all new options we add. Instead, the user only configured the options the user cars about.

Reply:
We could make some of the initial options as required (e.g. the two mentioned above), and future ones as optional... But I'm leaning a bit more to keep everything optional from the beginning.

Reply:
Ah I see. I'm in favor of making it optional.